cluster-autoscaler/main.go

```diff
@@ -403,7 +403,8 @@ func main() {
 					run(healthCheck, debuggingSnapshotter)
 				},
 				OnStoppedLeading: func() {
-					klog.Fatalf("lost master")
+					klog.Fatalf("lost master. Shutting down.")
+					klog.FlushAndExit(klog.ExitFlushTimeout, 1)
```

Comment:
It doesn't look to me like this is meant to be used *in addition* to `klog.Fatalf`:

```
// FlushAndExit flushes log data for a certain amount of time and then calls
// os.Exit. Combined with some logging call it provides a replacement for
// traditional calls like Fatal or Exit.
```

In fact this is unreachable code as `klog.Fatalf` has the outcome of calling `OsExit(255)`.

```
// Fatalf logs to the FATAL, ERROR, WARNING, and INFO logs,
// including a stack trace of all running goroutines, then calls OsExit(255).
// Arguments are handled in the manner of fmt.Printf; a newline is appended if missing.
func Fatalf(format string, args ...interface{}) {
	logging.printf(severity.FatalLog, logging.logger, logging.filter, format, args...)
}
```

Reply:
I don't know if it matters, but klog.Fatal does exit, see: https://github.com/kubernetes/klog/blob/e7125f792ea66a85818cfb45261c9e1acc585344/klog.go#L936-L964

Here's a test:
```
package main

import (
	"flag"

	"k8s.io/klog/v2"
)

func main() {
	klog.InitFlags(nil)
	flag.Parse()

	defer klog.Flush()

	klog.Info("About to encounter a fatal error...")
	klog.Fatalf("This is a fatal error - the application will exit here!")

	// This line will never be reached
	klog.Info("This message will never be printed")
}
```

```console
$ go run main.go
I0728 16:20:33.182323   99101 main.go:15] About to encounter a fatal error...
F0728 16:20:33.182456   99101 main.go:16] This is a fatal error - the application will exit here!
exit status 255
```